File: atcoder/abc151C_welcome_to_atcoder.py
# unihernandez22
# https://atcoder.jp/contests/abc151/tasks/abc151_c
# implementation

class Counter(dict):
    def __missing__(self, i):
        return 0

class BoolDict(dict):
    def __missing__(self, i):
        return False

n, m = map(int, input().split())

passed = 0
penalties = Counter()

ac = BoolDict()

for i in range(m):
    p, s = input().split()
    if not ac[p]:
        if s == "WA": penalties[p] += 1
        elif s == "AC":
            passed += 1
            ac[p] = True

x = 0
for i in penalties:
    if ac[i]:
        x += penalties[i]

print(passed, x)

File: atcoder/abc151C_welcome_to_atcoder.nim
# vicfred
# https://atcoder.jp/contests/abc151/tasks/abc151_c
# data structures, maps, implementation
import strutils, sequtils, tables

var n, m: int
(n, m) = stdin.readLine.split.map parseInt

var ac = initTable[int, int]()
var wa = initTable[int, int]()

var passed = 0
for i in 1..m:
  var x, s: string
  (x, s) = stdin.readLine.split

  let p = x.parseInt

  if p notin ac:
    if s == "WA":
      wa[p] = wa.getOrDefault(p) + 1
    else:
      passed += 1
      ac[p] = 1

var penalties = 0
for x in ac.keys:
  penalties += wa.getOrDefault(x)

echo passed," ", penalties

